### DISCUSSION TITLE
Standalone actions are not logged in actions events log

### DISCUSSION THREAD

**Discussion by muni-oscar-parra** [laravel/nova-issues#4697] [Q&A] [answered]
- Laravel Version: ^9.10
- Nova Version: ^4.8
- PHP Version: 8.1.4
- Database Driver & Version: mysql 5.7
- Operating System and Version: #.#
- Browser type and version: #.#
- Reproduction Repository: https://github.com/###/###

### Detailed steps to reproduce the issue on a fresh Nova installation:

- Create Standalone action
- Run Standalone action
- Go to nova/resources/action-events

**Comment by crynobone** [comment, marked as answer]
Action Events need to be attached to a target model and standalone actions don't select and target any model, therefore there are no action events for standalone actions and you need to create them manually if you wish to have them.

**Comment by hivokas** [comment]
@crynobone while I understand your reasoning, I think it would be great to see action events for standalone actions out of the box. Without that, I can't see who and when runs standalone actions.

**Reply by crynobone** [reply]
You can always manually create an Action Event using `Laravel\Nova\Actions\ActionEvent` model.

**Reply by hivokas** [reply]
@crynobone thanks for replying. The goal was to automatically log action events for standalone actions. I was able to achieve this by extending `ActionEvent` model:

```php
class ActionEvent extends \Laravel\Nova\Actions\ActionEvent
{
    public static function createForModels(ActionRequest $request, Action $action, $batchId, Collection $models, $status = 'running')
    {
        // If models collection is empty, we are dealing with a standalone action.
        // Standalone actions are not logged out of the box, but we want to log them.
        // https://github.com/laravel/nova-issues/discussions/4697
        if ($models->isEmpty()) {
            static::insert(
                static::defaultAttributes($request, $action, $batchId, $status),
            );
        }

        parent::createForModels($request, $action, $batchId, $models, $status);
    }
}
```
